py/qqa/plots/plotimage.py

```diff
@@ -0,0 +1,149 @@
+    fig.y_range.end = ny
+
+    if title is not None:
+        fig.title.text = title
+
+    return fig
+
+def main(input_in = None, output_in = None, downsample_in = None):
+
+    # if run from command line
+    if [input_in, output_in, downsample_in] == [None, None, None]:
+        import argparse
+        parser = argparse.ArgumentParser(usage = "{prog} [options]")
+        parser.add_argument("-i", "--input", type=str, required=True, help="input image fits file")
+        parser.add_argument("-o", "--output", type=str, help="output html file")
+        parser.add_argument("-d", "--downsample", type=int, default=4, help="downsample image NxN")
+
+        args = parser.parse_args()
+
+        if args.output is None:
+            base, ext = os.path.splitext(args.input)
+            assert ext == '.fits'
+            args.output = base + "-" + str(args.downsample) + 'x.html'
+
+        image = args.input
+        basename = os.path.basename(args.input)
+        n = args.downsample
+        output = args.output
+
+    # if run from a different file and has provided arguments
+    else:
+        if input_in == None or downsample_in == None:
+            return "input_in and/or downsample_in not provided"
+        if output_in is None:
+            base, ext = os.path.splitext(input_in)
+            assert ext == '.fits'
+            output_in = base + "-" + str(n_in) + 'x.html'
+        image = input_in
+        basename = os.path.basename(input_in)
+        n = downsample_in
+        output = output_in
+
+    image = fits.getdata(image)
+
+    short_title = '{basename} {n}x{n}'.format(basename=os.path.splitext(basename)[0], n=n)
+    long_title = '{basename} downsampled {n}x{n}'.format(basename=basename, n=n)
+
+    fig = plot_image(image, downsample=n, title=long_title)
+    plot_script, plot_div = components(fig)
+    html = """
+    <html>
+    <link
+        href="https://cdn.pydata.org/bokeh/release/bokeh-{version}.min.css"
+        rel="stylesheet" type="text/css"
+    >
+    <link
+        href="https://cdn.pydata.org/bokeh/release/bokeh-tables-{version}.min.css"
+        rel="stylesheet" type="text/css"
+    >
+    <script
+        src="https://cdn.pydata.org/bokeh/release/bokeh-{version}.min.js"
+    ></script>
+
+    <script src="https://cdn.pydata.org/bokeh/release/bokeh-tables-{version}.min.js"
+    ></script>
+    <body>
+    <div>{plot_script} {plot_div}</div>
+    <div>Downsample: <input size="4" type="number" id = "down" value={downsample}></div>
+    <button onclick='
+        var url = "{basename}-" + document.getElementById("down").value + "x.html"
+        window.open(url, "_top")
+    '>Generate</button>
+    </body>
+    <script>
+        var input = document.getElementById("down");
+            input.addEventListener("keyup", function(event) {{
+                if (event.keyCode === 13) {{
+                    event.preventDefault();
+            var url = "{cam}-" + document.getElementById("down").value + "x.html"
```

Comment:
Almost there...  I'm getting an error because the `{cam}` in this format string isn't included in the options to `.format()` at the end.

Also: could this be structured to keep the HTML and javascript within a separate jinja template file instead of moving it back into the python code?  This is different from how we did it for surveyqa, but I think it is cleaner to keep the HTML templates in separate files.